### ISSUE TITLE
False positive: Regex check 

### ISSUE THREAD

**Issue by felixLam** [stil4m/elm-analyse#66]
The regex check falsely identifies the following regex builder as "non-static", which is correct but might be considered a false positive for dynamic regex patterns:

```elm
regexStringComment : String -> Regex
regexStringComment key =
    Regex.regex ("\\{-\\| ([^\\}]*)\\n-\\}\\n" ++ key ++ "\\s+:")
```

**Comment by stil4m** [comment]
I believe this is not a false positive: When `regexStringComment "[+"` is invoked you will get a runtime exception.

**Comment by felixLam** [comment]
Ok, then I misread the analyser warning. I thought the issue is about performance not safety. Maybe there should be two different warnings: one if regex is used without static where the pattern is not dynamic and another warning about potential sanitization issues.

**Comment by stil4m** [comment]
Check. Good point. Can you create a new issue for the potential sanitisation?

**Comment by felixLam** [comment]
Not really clear how you would check that the string is correctly sanitized.

**Comment by stil4m** [comment]
I'll think about this. Should be possible to check this. I'll close this ticket though.
